app/src/main/java/ch/epfl/cs311/wanderwave/ui/screens/BeaconScreen.kt

```diff
@@ -89,12 +97,12 @@ private fun BeaconScreenPreview() {
 }
 
 @Composable
-private fun BeaconScreen(beacon: Beacon) {
+private fun BeaconScreen(beacon: Beacon, viewModel: BeaconViewModel = hiltViewModel()) {
```

Comment:
My idea with the BeaconScreen here was that it does not need the ViewModel passed, so we can use the @Preview (which does not work with Hilt, I think). 
I don't really know how many parameters it'll be, but if it's not too many, I suggest creating the functions that happen when adding a beacon here, and passing them in as callbacks, instead of passing in the ViewModel. 

If you do find a way to make the preview work with the ViewModel, you could also remove the `beacon` parameter and just get it from the ViewModel